great_expectations/core/usage_statistics/anonymizers/checkpoint_run_anonymizer.py

```diff
@@ -3,60 +3,92 @@
 from numbers import Number
 from typing import Any, Dict, List, Optional, Union
 
-import great_expectations.exceptions as ge_exceptions
-from great_expectations.checkpoint.util import (
-    get_substituted_validation_dict,
-    get_validations_with_batch_request_as_dict,
-)
+from ruamel.yaml.comments import CommentedMap
+
+from great_expectations.checkpoint.checkpoint import Checkpoint
 from great_expectations.core import RunIdentifier
 from great_expectations.core.batch import (
     BatchRequest,
     RuntimeBatchRequest,
     get_batch_request_as_dict,
 )
-from great_expectations.core.usage_statistics.anonymizers.anonymizer import Anonymizer
+from great_expectations.core.usage_statistics.anonymizers.base import BaseAnonymizer
 from great_expectations.core.usage_statistics.anonymizers.types.base import (
     CHECKPOINT_OPTIONAL_TOP_LEVEL_KEYS,
 )
-from great_expectations.core.util import get_datetime_string_from_strftime_format
+from great_expectations.data_context.types.base import CheckpointConfig
 from great_expectations.util import deep_filter_properties_iterable
 
 logger = logging.getLogger(__name__)
 
 
-class CheckpointRunAnonymizer(Anonymizer):
-    def __init__(self, salt=None):
-        super().__init__(salt=salt)
+class CheckpointRunAnonymizer(BaseAnonymizer):
+    def anonymize(self, obj: object = None, **kwargs) -> Any:
+        assert self.can_handle(
+            obj=obj
+        ), "CheckpointRunAnonymizer can only handle objects of type Checkpoint or CheckpointConfig"
+        if "config" in kwargs:
+            return self._anonymize_checkpoint_info(**kwargs)
+        return self._anonymize_checkpoint_run(obj=obj, **kwargs)
 
-        self._salt = salt
+    def _anonymize_checkpoint_info(self, name: str, config: dict) -> dict:
```

Comment:
Just an optional suggestion - could this be `_anonymize_checkpoint_config()` just to be more descriptive that it acts on checkpoint config dicts?

Reply:
I agree - done!